src/communicator.h

```diff
@@ -52,6 +52,12 @@ virtual void start() = 0;
  */
 virtual void stop() = 0;
 
+/**
+ * If the communicator uses tag internally, this method provides a way for the user to get a new tag
+ * to avoid conflicts.
```

Comment:
This comment is conflicting with the one on line 100. My understanding is tags above 1024 are reserved and should not be used as a user tag (what it says on line 100), but this comment says `use_new_tag` is a way for the user to get a new tag to avoid conflicts - so I interpret it as tags above 1024 will be used as user tags. Maybe the "user" term is overloaded here, so we should clarify it.

Reply:
This method is removed in 4f7f099 because buffer communicator switched back to grouping by column, which means a GPU does not need to send multiple messages to the same remote GPU.